src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -594,7 +625,7 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                     // response, since it's not a valid parameter when using the implicit flow.
                     // See http://openid.net/specs/openid-connect-core-1_0.html#Authentication
                     // and https://tools.ietf.org/html/rfc6749#section-4.2.2.
-                    SaveTokens(ticket.Principal, message, saveRefreshToken: false);
+                    SaveTokens(ticket.Principal, message, jwt.Issuer, saveRefreshToken: false);
```

Comment:
@vibronet @Tratcher I understand that logically the issuer of the artifacts is the issuer. However, it is not true that the IDP issued those claims. Setting the issuer == 'iss' is implying the claims arrived inside a signed token. Don't know what the harm is, but we are deviating from what arrived. I am not aware how the consumer will be consume these claims.